frontend-new/src/consent/components/consentPage/Consent.tsx

```diff
@@ -73,8 +71,18 @@ const DataProtectionAgreement: React.FC = () => {
       };
       setIsAcceptingDPA(true);
       const prefs = await userPreferencesService.updateUserPreferences(newUserPreferenceSpecs);
-      userPreferencesStateService.setUserPreferences(prefs);
-      navigate(routerPaths.ROOT, { replace: true });
+
+      userPreferencesStateService.setUserPreferences({
+        ...prefs,
+        sensitive_personal_data_requirement: userPreferences?.sensitive_personal_data_requirement!,
```

Comment:
No, it is not undefined because.

get user preferences adds this flag because it indicates weather you need to provide sensitive personal data requirement or not.

After logging in we fetch your preferences and we already know if you have to provide the sensitive personal data or not. 

So I would say it is not undefined.

If you've created a new account you need to log in first.

So either way we are going to have this field defined here.